### PR TITLE
Add select all buttons to change dialog

### PR DESCRIPTION
**Pull request by Siedlerchr**
when merging entries
Fixes https://github.com/koppor/jabref/issues/385

grafik

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for bigger UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
After talk with @koppor 
grafik

Use all current changes from JabRef = do nothing = cancel
select all changes from jabref = uncheck all checkboxes and set all radiobuttons in merge to left
select all changes from disk = check all checkboxes and set all radiiobuttons to right

However, I still need to find a proper binding way to hide the checkbox in the merge entries when switching between the merge entries and added/deleted

**Comment by koppor** [comment]
I would group the buttons "Select ..." together (on the left), then a small dividing space and then the other two buttons which will lead to something really changing (including a close of the dialog).

**Comment by tobiasdiez** [comment]
I find the dialog with 6 new buttons relatively complex. As the user it's not really clear what is meant with "accept" vs "select". Personally, I prefer the dichotomy between accept & dismiss changes. Especially, "changes from JabRef" is somewhat misleading as the only changes are on the disk relative to the currently in-memory version. 

Moreover, selection of all changes in the entry diff dialog is perhaps a bit too advanced to be placed that prominently. What about adding the functionality to the left / right arrows at the top?

**Comment by koppor** [comment]
2nd point with arrows and selection: +1

Regarding the plenty of buttons. I'm unsure. Maybe, it just helps to move the selection buttons below the list of changes.

I know that a design like the merge dialog in IntelliJ would be better. However, we currently don't have the man power for that...

**Comment by Siedlerchr** [comment]
grafik

I incorporated the feedback and made now the arrows clickable for select all radio buttons (added tooltip) 
and I moved the buttons mark and unmark to the left

File: src/main/java/org/jabref/gui/collab/ChangeDisplayDialog.java [hunk 1]
```diff
@@ -21,47 +25,68 @@
...
+
+        pane.getItems().addAll(leftScroll, infoPanel);
+        pane.setResizableWithParent(leftScroll, false);
+
+        getDialogPane().setContent(pane);
 
         Label rootInfo = new Label(Localization.lang("Select the tree nodes to view and accept or reject changes") + '.');
         infoPanel.setCenter(rootInfo);
-        tree.getSelectionModel().select(0);
+        changesList.getSelectionModel().select(0);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/collab/ChangeDisplayDialog.java]: I think there is a `selectFirst` method.

File: src/main/java/org/jabref/gui/collab/EntryChangeViewModel.java [hunk 2]
```diff
@@ -39,6 +41,19 @@ public EntryChangeViewModel(BibEntry entry, BibEntry newEntry, BibDatabaseContex
 
     }
 
+    /**
+     *   We override this here to select the radio buttons accordingly
+     */
+    @Override
+    public void setAccepted(boolean accepted) {
+        super.setAccepted(accepted);
+        if (accepted) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/collab/EntryChangeViewModel.java]: This is no longer needed now, right?

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/collab/EntryChangeViewModel.java]: Nope, this is still needed. It ensures that when you click on "mark all changes" that the radio buttons on the right are selected and the merge entry will be the correct one.

**Pull request merged**